canonical-web-and-design/vanilla-framework
# Add application layout with examples and docs

## Done

Adds application layout and panel component with documentation and examples.

## QA

- `./run` or [demo](https://vanilla-framework-3175.demos.haus/docs/examples/layout/application)
- Check application layout documentation: https://vanilla-framework-3175.demos.haus/docs/layouts/application
  - please note that CodePen will show unstyled examples (as this version is not published yet)
- Check [Application layout structure example](https://vanilla-framework-3175.demos.haus/docs/examples/layout/application-structure)
     - verify it behaves as expected on all screen sizes
- Check full [Application layout example](https://vanilla-framework-3175.demos.haus/docs/examples/layout/application)
     - verify it behaves as expected on all screen sizes

## Screenshots

### Structure

<img width="1316" alt="Screenshot 2020-08-04 at 12 07 32" src="https://user-images.githubusercontent.com/83575/89282288-98a40a00-d64b-11ea-9b47-dd1362eca7ac.png">

### Small screen

<img width="667" alt="Screenshot 2020-08-04 at 12 08 42" src="https://user-images.githubusercontent.com/83575/89282282-9772dd00-d64b-11ea-897a-ac71be5a206b.png">

### Medium screen

<img width="1019" alt="Screenshot 2020-08-04 at 12 08 28" src="https://user-images.githubusercontent.com/83575/89282283-980b7380-d64b-11ea-8ea4-00d678f9e419.png">

### Large screen

<img width="1315" alt="Screenshot 2020-08-04 at 12 07 53" src="https://user-images.githubusercontent.com/83575/89282286-98a40a00-d64b-11ea-8f45-1594726bd1b8.png">

### Tasks:

- [x] move is-pinned class to sidenav instead of top level layout
- [x] ~~make side nav 'mobile 'toggle' bar' part of side navigation panel~~ (not possible without additional wrapper markup)
- [x] overlay and aside should be one panel, just different variant (pinned/unpinned)
- [x] separate styling classes into consistent child panel element
  - make sure it works with small amount of content (nothing to scroll) and with lot of content (scroll)
  - move border and shadow to layout styling from panel
- [x] remove any unnecessary demo styling
- [x] create panel component based on existing demo styling
- [x] create minimal layout demo (just layout, no content)
- [x] fix demo side navigation include (to avoid docs side navigation drawer)
- [x] fix sidebar panel scrolling issue (side nav scrolls and whole panel scrolls)
- [x] fix stacked form demo HTML (includes)
- [x] add application layout documentation
- [x] remove styling from layout
- [x] separate file for panel styling
